xarray/core/dataset.py

```diff
@@ -1082,7 +1088,34 @@ def _construct_direct(
         if dims is None:
             dims = calculate_dimensions(variables)
         obj = object.__new__(cls)
-        obj._variables = variables
+        obj._manifest = DataManifest(variables=variables)
+        obj._variables = obj._manifest.variables
+        obj._coord_names = coord_names
+        obj._dims = dims
+        obj._indexes = indexes
+        obj._attrs = attrs
+        obj._close = close
+        obj._encoding = encoding
+        return obj
+
+    @classmethod
+    def _construct_from_manifest(
+        cls,
+        manifest,
+        coord_names,
+        dims=None,
+        attrs=None,
+        indexes=None,
+        encoding=None,
+        close=None,
+    ):
+        """Creates a Dataset that is forced to be consistent with a DataTree node that shares its manifest."""
```

Comment:
The idea is to use this when someone calls `dt.ds` - so that the `DataTree` returns an actual `Dataset`, but one whose contents are linked to the wrapping tree node.